### ISSUE TITLE
Deserialize object inside properties

### ISSUE THREAD

**Issue by HarelM** [NetTopologySuite/NetTopologySuite.IO.GeoJSON#45]
I'm not sure if this is possible or not, and if this requires some implementation.
Here's the gist of it:
I want to set an object inside the attributes table (properties) and get it back after serialization and deserialization.
Something like the following test:
```cs
        [TestMethod]
        public void Test()
        {
            var feature = new Feature(new Point(0, 0), new AttributesTable
            {
                { "data", new MyClass { MyProp = new MyProp { A = new DateTime()}} }
            });
            var bytes = new FeatureCollection { feature }.ToBytes(); //serialize
            var coll = bytes.ToFeatureCollection(); // deserialize
            Assert.IsTrue(coll[0].Attributes["data"] is MyClass);
        }
```

**Comment by airbreather** [comment]
Got a full repro that applies on both 1.15.5 and 2.0.1:

# ConsoleApp0.csproj
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp3.0</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="NetTopologySuite.IO.GeoJSON" Version="2.0.1" />
    <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
  </ItemGroup>

</Project>
```

# Program.cs
```csharp
using System;
using System.IO;
using System.Text;

using NetTopologySuite.Features;
using NetTopologySuite.Geometries;
using NetTopologySuite.IO;

using Newtonsoft.Json;

static class Program
{
    static void Main()
    {
        var js = GeoJsonSerializer.Create();
        js.TypeNameHandling = TypeNameHandling.Objects;
        js.Formatting = Formatting.Indented; // to help debugging

        var feature = new Feature(new Point(0, 0), new AttributesTable
        {
            { "data", new MyClass() },
        });

        using var ms = new MemoryStream();

        using (var sw = new StreamWriter(ms, Encoding.UTF8, 1024, true))
        using (var wr = new JsonTextWriter(sw))
        {
            js.Serialize(wr, feature);
        }

        ms.Position = 0;

        string serializedText = Encoding.UTF8.GetString(ms.ToArray()); // to help debugging

        using (var sr = new StreamReader(ms, Encoding.UTF8, false, 1024, true))
        using (var rd = new JsonTextReader(sr))
        {
            feature = js.Deserialize<Feature>(rd);
        }

        // these two should write the same thing, but they don't in 2.0.1 or 1.15.5
        Console.WriteLine(typeof(MyClass));
        Console.WriteLine(feature.Attributes["data"].GetType());
    }

    class MyClass
    {
    }
}
```

**Comment by FObermaier** [comment]
I implemented a fix for this. You need to set the `JsonSerializer.TypeNameHandling` to `TypeNameHandling.Objects` to make it work:
https://github.com/NetTopologySuite/NetTopologySuite.IO.GeoJSON/blob/f8d7fbe62ce9e31a11a0144e8ac618d23dc2d641/test/NetTopologySuite.IO.GeoJSON.Test/Issues/NetTopologySuite.IO.GeoJSON/GitHubIssues.cs#L423-L451

**Comment by HarelM** [comment]
Nice! Thanks!

**Comment by airbreather** [comment]
Fix has finally been released in 2.0.3, closing.
